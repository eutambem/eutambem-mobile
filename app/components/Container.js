import React from 'react';
import PropTypes from 'prop-types';
import { SafeAreaView, KeyboardAvoidingView } from 'react-native';

import appStyles from '../styles';

const Container = props => (
  <SafeAreaView style={appStyles.container}>
    <KeyboardAvoidingView
      style={appStyles.innerContainer}
      keyboardVerticalOffset={60}
      behavior="position"
    >
      {props.children}
    </KeyboardAvoidingView>
  </SafeAreaView>
);

Container.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.element),
    PropTypes.element,
  ]).isRequired,
};

export default Container;
