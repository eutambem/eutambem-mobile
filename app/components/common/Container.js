import React from 'react';
import PropTypes from 'prop-types';
import { SafeAreaView, ScrollView } from 'react-native';

import appStyles from '../../styles';

const Container = props => (
  <SafeAreaView style={appStyles.container}>
    <ScrollView style={[appStyles.innerContainer, props.innerContainerStyle]}>
      {props.children}
    </ScrollView>
  </SafeAreaView>
);

Container.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.element),
    PropTypes.element,
  ]).isRequired,
  innerContainerStyle: PropTypes.oneOfType([
    PropTypes.number,
    PropTypes.object,
  ]),
};

Container.defaultProps = {
  innerContainerStyle: {},
};

export default Container;
