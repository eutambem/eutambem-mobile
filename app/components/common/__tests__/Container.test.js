import React from 'react';
import { Text } from 'react-native';
import { shallow } from 'enzyme';

import Container from '../Container';

describe('Container', () => {
  it('renders correctly', () => {
    const component = (
      <Container>
        <Text>Testing Container</Text>
      </Container>
    );
    const wrapper = shallow(component);
    expect(wrapper).toMatchSnapshot();
  });
});
