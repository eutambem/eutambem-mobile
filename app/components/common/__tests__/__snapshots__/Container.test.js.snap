// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container renders correctly 1`] = `
<SafeAreaView
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flex": 1,
    }
  }
>
  <KeyboardAvoidingView
    behavior="position"
    enabled={true}
    keyboardVerticalOffset={50}
  >
    <ScrollView
      style={
        Array [
          Object {
            "padding": "5%",
          },
          Object {},
        ]
      }
    >
      <ForwardRef(Text)>
        Testing Container
      </ForwardRef(Text)>
    </ScrollView>
  </KeyboardAvoidingView>
</SafeAreaView>
`;
